server/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -268,7 +267,8 @@
         4096,
         1,
         Integer.MAX_VALUE,
-        Setting.Property.NodeScope
+        Setting.Property.NodeScope,
+        Setting.Property.DeprecatedWarning
```

Comment:
When would we raise a deprecation warning for node settings, only on startup?

Reply:
Have confirmed with @pgomulka that this will only be emitted on startup, yes.